torch/_dynamo/compiled_autograd.py

```diff
@@ -0,0 +1,176 @@
+import contextlib
+import functools
+from typing import List
+
+import torch
+from torch._dynamo.external_utils import call_hook
+from torch._dynamo.source import GetItemSource, LocalSource
+from torch._dynamo.utils import counters
+from torch._subclasses import FakeTensorMode
+from torch.fx import GraphModule
+from torch.fx.experimental.proxy_tensor import (
+    decompose,
+    disable_autocast_cache,
+    disable_proxy_modes_tracing,
+    fetch_tensor_proxy,
+    ProxyTorchDispatchMode,
+    PythonKeyTracer,
+    track_tensor_tree,
+)
+from torch.fx.experimental.symbolic_shapes import DimDynamic, ShapeEnv
+
+
+class AutogradCompilerInstance:
+    def __init__(self, compiler_fn) -> None:
+        self.compiler_fn = compiler_fn
+        self.stack = contextlib.ExitStack()
+        self.shape_env = ShapeEnv()
```

Comment:
Do you need a fresh one? I don't 100% understand the lifecycle of when this is made, so I hesitate suggesting taking it off the TracingContext, but if this is fully subsumed by a PT2 eval_frame, you might be able to reuse that one?

Reply:
We can't reuse the prior compile time `ShapeEnv()` because there is no mapping between the concreate sizes stored in C++ and the symbolic sizes where they came from.  At this point we just get an `autograd::Node` has a size variable `foo` which equals 10.